### DISCUSSION TITLE
Forbid/warn against connections and new widgets with incorrect/incompatible types - by Mr. Guillen (@arthexis)

### DISCUSSION THREAD

**Discussion by KennedyRichard** [IndiePython/nodezator#47] [Feature Request]
Feature title (updated): **Forbid/warn against connections and new widgets with incorrect/incompatible types**

Proponent: Mr. Guillen (@arthexis)

Added element from Discord user Croissant: new widgets in a node (check update on the bottom of this message)

Hello, everyone,

I'm creating this discussion just to notify everybody of the inclusion of this feature in the list of requested/planned features (#13) and also, naturally, as a place dedicated to present and future discussions/feedback on the feature, specially when it is time to schedule it for implementation.

The idea was first mentioned in this [comment in another discussion](https://github.com/IndiePython/nodezator/discussions/45#discussioncomment-5561580), the contents of which are reproduced below:

> I had this idea: how hard would you think it'd be to implement something
that notifies the user if a connection they are making is somehow illegal
(ie, like if you get a mypy typing warning) ?

I provided further comment on the idea in [this comment](https://github.com/IndiePython/nodezator/discussions/45#discussioncomment-5585034).

**2023-June-21 update**: updated discussion title and feature title to also consider new widgets as elements whose data types must be forbidden/warned against, since nodes in Nodezator can also receive data via widgets added after the node's creation in variable parameters like `*args` and `**kwargs`. This update was a suggestion from a user named Croissant in [this Discord message](https://discord.com/channels/980193047000272967/980193047461630113/1121166565166022727).

**Comment by KennedyRichard** [comment]
Updated discussion title and feature title to include suggestion from Discord user Croissant, as described in the top message of this discussion.

**Comment by GrimPixel** [comment]
Some additional context: dimensions.
https://en.wikipedia.org/wiki/Dimensional_analysis#Formulation
